certbot/main.py

```diff
@@ -388,6 +389,85 @@ def _init_le_client(config, authenticator, installer):
 
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
+def _delete_if_appropriate(config): # pylint: disable=too-many-locals,too-many-branches
+    """Does the user want to delete their now-revoked certs?"""
+    display = zope.component.getUtility(interfaces.IDisplay)
+    reporter_util = zope.component.getUtility(interfaces.IReporter)
+
+    if config.namespace.noninteractive_mode:
```

Comment:
Removing this revealed that some of the integration tests were actually broken (because they all run with -n), but they've now been fixed.